### ISSUE TITLE
Incompatible with Talos 1.5 and 1.6 due to resource API changes

### ISSUE THREAD

**Issue by eugene-marchanka** [siderolabs/talos-vmtoolsd#9]
vmtools does not export node IP in Talos `v1.5.1`

```
$ kubectl --kubeconfig kubeconfig logs
 -n kube-system talos-vmtoolsd-f5bzt                                                                              
{"level":"info","msg":"talos-vmtoolsd version 0.3.1\nCopyright 2020-2022 Oliver Kuckertz <oliver.kuckertz@mologie.
de>\nThis program is free software and available under the Apache 2.0 license."}                                  
{"error":"rpc error: code = Unimplemented desc = unknown service resource.ResourceService","level":"error","module
":"talosapi","msg":"error receiving address status resource"}                                                     
{"error":"rpc error: code = Unimplemented desc = unknown service resource.ResourceService","level":"error","module
":"talosapi","msg":"error receiving address status resource"}                                                     
{"level":"warning","module":"tboxcmds","msg":"not sending primary IP: no interfaces received from upstream"}      
{"error":"rpc error: code = Unimplemented desc = unknown service resource.ResourceService","level":"error","module
":"talosapi","msg":"error receiving address status resource"}                                                     
{"error":"rpc error: code = Unimplemented desc = unknown service resource.ResourceService","level":"error","module
":"talosapi","msg":"error receiving address status resource"}                                                     
{"level":"warning","module":"tboxcmds","msg":"not sending primary IP: no interfaces received from upstream"}      
{"error":"rpc error: code = Unimplemented desc = unknown service resource.ResourceService","level":"error","module
":"talosapi","msg":"error receiving address status resource"}                                                     
{"error":"rpc error: code = Unimplemented desc = unknown service resource.ResourceService","level":"error","module
":"talosapi","msg":"error receiving address status resource"}                                                     
{"level":"warning","module":"tboxcmds","msg":"not sending primary IP: no interfaces received from upstream"}      
{"error":"rpc error: code = Unimplemented desc = unknown service resource.ResourceService","level":"error","module
":"talosapi","msg":"error receiving address status resource"}                                                     
{"error":"rpc error: code = Unimplemented desc = unknown service resource.ResourceService","level":"error","module
":"talosapi","msg":"error receiving address status resource"}                                                     
{"level":"warning","module":"tboxcmds","msg":"not sending primary IP: no interfaces received from upstream"}      
{"error":"rpc error: code = Unimplemented desc = unknown service resource.ResourceService","level":"error","module
":"talosapi","msg":"error receiving address status resource"}
```

**Comment by mologie** [comment]
Talos 1.5 removed an API for listing resources that talos-vmtoolsd relied on. I have a branch now where the new API is being used, but it is entirely untested at  this point. It might not launch, crash on launch, or report wrong data:

To install, please use the "unstable" yaml but insert image reference `ghcr.io/mologie/talos-vmtoolsd-unstable:talos-1.5`

**Comment by eugene-marchanka** [comment]
New image worked!
![image](https://github.com/mologie/talos-vmtoolsd/assets/81696153/bc74d606-6442-4f25-b9b1-7d5d55e4efc3)

**Comment by mologie** [comment]
Great, thanks for the info! I will leave this open for visibility for others, and close with an official stable release.

**Comment by xerxist** [comment]
> Talos 1.5 removed an API for listing resources that talos-vmtoolsd relied on. I have a branch now where the new API is being used, but it is entirely untested at this point. It might not launch, crash on launch, or report wrong data:
> 
> To install, please use the "unstable" yaml but insert image reference `ghcr.io/mologie/talos-vmtoolsd-unstable:talos-1.5`

This works for me 2.

Cheers!

**Comment by CompPhy** [comment]
Considering that Talos 1.6 just went GA last week, maybe it's time to get this merged???

I've been running 1.5.x for a while and started testing vsphere-csi-driver recently.  This driver relies on cloud-provider-vsphere, which uses the nic/IP information to make sure it's matching up nodes correctly in vCenter.  Don't ask me why it works this way, I just know that it's causing weird errors because the IP isn't populated in vCenter.

Basically, without working vmtools it can cause other problems downstream.  I just spent several days pulling my hair out until I realized that this is a known, and fixed, issue.... Except you have to explicitly come here to find the fix.

If nothing else, maybe add a note on the README.md for anyone that is using the Talos 1.5 or 1.6 release???

**Comment by CompPhy** [comment]
FWIW, I can confirm this fix works on Talos 1.5.5; it also immediately fixed my issues with cloud-provider-vsphere as soon as the IP populated correctly in vCenter.

**Comment by jonkerj** [comment]
@mologie, any news on this matter?

**Comment by mologie** [comment]
Hi folks, sorry for the late response and especially inconveniences caused. Family and life happened, and this is unfortunately a project that does not get much time allocated at work. If there is a commercial sponsor that would like to take over maintenance (like previously discussed on Talos Slack) that would be great for this project.

Given the Broadcom licensing changes at VMware, Authentic Vision will migrate away from VMware entirely this year, so I would have to make this a personal project of mine. I am not sure I am willing to commit to that since the customer base is entirely business customers. If anything, this would be a portfolio-project.

Regardless, I'll see to polish up, test, and get a release ready (including changes in preparation for making this a machine extension, but no extension available yet!) this week.

**Comment by jonkerj** [comment]
In the meanwhile, we (Equinix) have offered to adopt te project. While discussing whether to transfer the repo as a whole or change permissions, the idea came up to ask Sidero if this project could be hosted under their org. They were enthusiastic, and so that happened. 

Equinix (primarily @lennardk, @robinelfrink and myself) will do most the actual work, though.

In the short term we are going to make the tool compatible with 1.6, and patch up @bnason's system extension (if needed). Our goal is to have the latter upstreamed at some point.

**Comment by mologie** [comment]
Thank you @jonkerj, @lennardk, and @robinelfrink for adopting this project. With 0.3.2 being ported to the Sidero group too everything and the work-in-progress stuff is available here, so it is time for me to step back and let you do your magic. I am of course available for questions and concerns regarding code and integration, but trust that the project is in good hands.

**Comment by omniproc** [comment]
any updates on this?

**Comment by robinelfrink** [comment]
Hi @omniproc,

Sorry for the delay. We have just released v0.4.0.
